ios/xcode.mjs

```diff
@@ -32,13 +44,148 @@ export const PRODUCT_VERSION = "PRODUCT_VERSION";
 export const USER_HEADER_SEARCH_PATHS = "USER_HEADER_SEARCH_PATHS";
 export const WARNING_CFLAGS = "WARNING_CFLAGS";
 
+/**
+ * @param {JSONObject} platformConfig
+ * @param {ProjectConfiguration} project
+ * @param {string} projectRoot
+ * @param {string} destination
+ * @returns {ProjectConfiguration}
+ */
+export function applyBuildSettings(
+  platformConfig,
+  project,
+  projectRoot,
+  destination,
+  fs = nodefs
+) {
+  const codeSignEntitlements = platformConfig["codeSignEntitlements"];
+  if (isString(codeSignEntitlements)) {
+    const appManifest = findFile("app.json", projectRoot, fs);
+    if (!appManifest) {
+      throw new Error("Cannot find 'app.json'");
+    }
+
+    const packageRoot = path.dirname(appManifest);
+    const entitlements = path.join(packageRoot, codeSignEntitlements);
+    const relPath = path.relative(destination, entitlements);
+    project.buildSettings[CODE_SIGN_ENTITLEMENTS] = relPath;
+  }
+
+  const codeSignIdentity = platformConfig["codeSignIdentity"];
+  if (isString(codeSignIdentity)) {
+    project.buildSettings[CODE_SIGN_IDENTITY] = codeSignIdentity;
+  }
+
+  const developmentTeam = platformConfig["developmentTeam"];
+  if (isString(developmentTeam)) {
+    project.buildSettings[DEVELOPMENT_TEAM] = developmentTeam;
+    project.testsBuildSettings[DEVELOPMENT_TEAM] = developmentTeam;
+    project.uitestsBuildSettings[DEVELOPMENT_TEAM] = developmentTeam;
+  }
+
+  const bundleIdentifier = platformConfig["bundleIdentifier"];
+  if (isString(bundleIdentifier)) {
+    project.buildSettings[PRODUCT_BUNDLE_IDENTIFIER] = bundleIdentifier;
+    project.testsBuildSettings[PRODUCT_BUNDLE_IDENTIFIER] =
+      `${bundleIdentifier}Tests`;
+    project.uitestsBuildSettings[PRODUCT_BUNDLE_IDENTIFIER] =
+      `${bundleIdentifier}UITests`;
+  }
+
+  const buildNumber = platformConfig["buildNumber"];
+  project.buildSettings[PRODUCT_BUILD_NUMBER] =
+    buildNumber && isString(buildNumber) ? buildNumber : "1";
+
+  return project;
+}
+
+/**
+ * @param {ProjectConfiguration} project
+ * @returns {void}
+ */
+export function applyPreprocessorDefinitions({
+  reactNativeVersion,
+  useNewArch,
+  useBridgeless,
+  buildSettings,
+}) {
+  const existing = buildSettings[GCC_PREPROCESSOR_DEFINITIONS];
+  const preprocessors = Array.isArray(existing) ? existing : [];
+
+  preprocessors.push(`REACT_NATIVE_VERSION=${reactNativeVersion}`);
+
+  // In Xcode 15, `unary_function` and `binary_function` are no longer provided
```

Comment:
I know this is a port, but maybe we wont' need this eventually as Xcode N - 1 support is quite quick to drop.

Reply:
It's the other way around. As long as we support older versions of React Native, we will need this workaround.